47.md

```diff
@@ -135,6 +137,95 @@ The **client** should then store this connection and use it when the user wants
+
+- `relay` Required. URL of the relay where the **client** intends to communicate with the **wallet service**. May be more than one.
+
+The **wallet service** MAY ignore all the below optional parameters.
+
+- `name`: Optional. The name of the client app.
+- `icon`: Optional. The URL of an icon of the client app to display on the confirmation page.
+- `return_to` Optional. URI to open after the connection is created.
+- `expires_at` Optional. The connection cannot be used after this date. Unix timestamp in seconds.
+- `max_amount` Optional. The maximum amount in millisats that can be sent per renewal period
+- `budget_renewal` Optional. The reset the budget at the end of the given budget renewal. Can be `never` (default), `daily`, `weekly`, `monthly`, `yearly`
+- `request_methods` Optional. The url encoded, space separated list of request types that you need permission for. For example: `..&request_methods=pay_invoice%20get_balance`
+- `notification_types` Optional. The url encoded, space separated list of notification types that you need permission for. For example: `..&notification_types=payment_received%20payment_sent`
+- `isolated` Optional. The makes an isolated app connection / sub-wallet with its own balance and only access to its own transaction list. e.g. `&isolated=true`.
+- `metadata` Optional. Url encoded, JSON-serialized metadata that describes the app connection.
+
+The **user** opens the URL in the **wallet service** by scanning a QR code, handling a deeplink or pasting in a URI, and MUST be presented with a confirmation page.
+
+If the **user** approves the request, a new connection should be created for the **client**'s public key (specified in the base path of the authorization URI). Once the connection is created, the NIP-47 info event MUST be broadcasted to the relays specified in the authorization URI, and the info event MUST include a `p` tag containing the public key of the **client** this info is for, so that the **client** can discover the public key of the **wallet service**. The `p` tag MAY contain the recommended relay url of the **wallet service**.
+
+The **client** is then able to subscribe to the NIP-47 info event (kind 13194) filtered by setting the `p` tag to the pubkey of the **client**. The **client** will then know the public key of the **wallet service** (as the publisher of the info event).
+
+At this point, both the **client** and **wallet service** have all information required to communicate over NWC.
+
+The `lud16` field cannot be immediately communicated via the NIP-47 event to avoid metadata leaks. To discover the lightning address, the **client** MAY look for the presence of the  `lud16` parameter in the `get_info` response.
+
+#### Example Authorization URI
+
+`nostr+walletauth://b889ff5b1513b641e2a139f661a661364979c5beee91842f8f0ef42ab558e9d4?relay=wss%3A%2F%2Frelay.example.com%2Fv1`
+
+To target a specific wallet, a `nostr+walletauth+walletname` scheme may be used.
```

Comment:
Any reason for specifying walletname in the protocol vs. as an additional query param or path component?

Reply:
I think that's due to the fact that on iOS if you have multiple apps registered to handle the same protocol it will just pick the first one (instead of letting the user choose). 

The same problem exists for `lightning:` links and the only viable solution so far seems to let users pick from a list of wallets and use the vendor-specific links instead of the generic one.

Reply:
Ok that makes sense thanks.